modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/provider/APIApplicationProviderImpl.java

```diff
@@ -136,7 +129,9 @@ public RetrieveType getRetrieveType() {
 							"retrieveType");
 
 						if (Objects.equals(
-								listEntry.getKey(), "singleElement")) {
+								listEntry.getKey(),
+								HeadlessBuilderConstants.
+									RETRIEVE_TYPE_SINGLE_ELEMENT)) {
```

Comment:
I am not sure if this is the best way to parse the value, as we are keeping the values in two different classes, one is the enum and another one is the String constants.

Maybe it is better to include a new attribute in the RetrieveType enum and a parse method:

```
public enum RetrieveType {

	COLLECTION("collection"), SINGLE_ELEMENT("singleElement");

	public static RetrieveType parse(String value) {
		for (RetrieveType retrieveType : RetrieveType.values()) {
			if (Objects.equals(retrieveType.getValue(), value)) {
				return retrieveType;
			}
		}

		throw new IllegalArgumentException("Invalid value " + value);
	}

	public String getValue() {
		return _value;
	}

	private RetrieveType(String value) {
		_value = value;
	}

	private final String _value;

}
```

And then, call the method:

```
return RetrieveType.parse(listEntry.getKey());
```

Reply:
I've never seen this, but I like it. I think that this makes sense for scopes one